### DISCUSSION TITLE
Creating and sending an A28 based on an errored A34

### DISCUSSION THREAD

**Discussion by jj2fowler** [nextgenhealthcare/connect#6076] [Q&A]
Hi all,

We have the following problem that we are attempting to solve in Mirth. The scenario is that an A34 Merge Patient message is sent to a destination system in which the target patient does not exist. A NACK similar to the following is received then this occurs.

NACK sent from receiver. (Error)
Error
^^^ADTA34&A Patient with MRN of 12345 does not exist, so unable to merge

What we would then like to do is construct an A28 New Patient message from the PID/MSH information in the original A34 and send that to the target destination. Finally, the original A34 message would then be sent to the same destination.

Would the best (?only) way to do this be via the postprocessor? I cannot see an easy way to construct a new message type using the data from another message via this option, however Resending the original message seems straight forward enough, though.

Does anyone have any experience with something like this and is there a better approach?

Regards,

Jeremy.

**Comment by kirbykn2** [comment]
I would lean towards creating another destination that convert the A34 to an A28 when a NAK is received.
